daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/CompiledPackages.scala

```diff
@@ -21,7 +21,7 @@ private[lf] abstract class CompiledPackages(
   def definitions: PartialFunction[SDefinitionRef, SDefinition] =
     Function.unlift(this.getDefinition)
 
-  final def compiler: Compiler = new Compiler(pkgInterface, compilerConfig)
+  final lazy val compiler: Compiler = new Compiler(pkgInterface, compilerConfig)
```

Comment:
As far I can tel , there is no state in the Compiler, so it should be safe to make it a `val`.